hazelcast/src/main/java/com/hazelcast/spi/OperationService.java

```diff
@@ -106,6 +106,20 @@
                                             Collection<Integer> partitions) throws Exception;
 
     /**
+     * Invokes an set of operation on selected set of partitions
+     * * <p/>
+     * This method blocks until all operations complete.
+     *
+     * @param serviceName      the name of the service
+     * @param operationFactory the factory responsible for creating operations
+     * @param partitions       the partitions the operation should be executed on.
+     * @return a Map with partitionId as key and the outcome of the operation as value.
+     * @throws Exception
+     */
+    Map<Integer, Object> invokeOnPartitions(String serviceName, OperationFactory operationFactory,
+                                            int[] partitions) throws Exception;
```

Comment:
I found no good way to convert an `int[]` to a `Collection<Integer>` and I didn't want to introduce litter again. Maybe we can get rid of the other method eventually, since the array should be more litter friendly.